src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -86,5 +86,13 @@ private protected IReadOnlyList<PackageReference> GetTransitivePackageReferences
                     .ToList();
             }
         }
+
+        /// <summary>
+        /// Return all targets (dependency graph) found in project.assets.json file
+        /// </summary>
+        /// <param name="token">Cancellation token</param>
+        /// <returns>A list, one element for each framework restored, or <c>null</c> if project.assets.json file is not found</returns>
+        /// <remarks>Projects need to be NuGet-restored before calling this function</remarks>
+        internal abstract Task<IList<LockFileTarget>> GetFullRestoreGraphAsync(CancellationToken token);
```

Comment:
Consider moving above logic from`LegacyPackageReferenceProject.cs` to here, so there would be no code duplication.

Reply:
Fixed.